application/shared-kernel/DomainCore/Identity/StronglyTypedUlid.cs

```diff
@@ -18,19 +18,41 @@ public static T NewId()
     [UsedImplicitly]
     public static bool TryParse(string? value, out T? result)
     {
-        var success = Ulid.TryParse(value ?? "", out var parsedValue);
+        if (value is null)
+        {
+            result = null;
+            return false;
+        }
+
+        var prefix = GetPrefix();
+
+        var idValue = !string.IsNullOrWhiteSpace(prefix) ? value.Replace($"{prefix}_", string.Empty) : value;
+
+        var success = Ulid.TryParse(idValue, out var parsedValue);
         result = success ? FormUlid(parsedValue) : null;
         return success;
     }
 
     private static T FormUlid(Ulid newValue)
     {
+        var prefix = GetPrefix();
         return (T)Activator.CreateInstance(
             typeof(T),
             BindingFlags.Instance | BindingFlags.Public,
             null,
-            [newValue.ToString()],
+            [!string.IsNullOrWhiteSpace(prefix) ? $"{prefix}_{newValue}" : newValue.ToString()],
             null
         )!;
     }
+
+    private static string GetPrefix()
+    {
+        var attributes = typeof(T).GetCustomAttributes(typeof(WithIdPrefixAttribute), true);
+        if (attributes.Length == 0)
+        {
+            return string.Empty;
```

Comment:
I think we should not allow IDs without prefix's. At least i cannot think of a scenario where it will be useful, and I fear that developers will unknowingly create new IDs without a prefix. If they are truly needed, we could allow the Prefix to be `""`

Reply:
Hm I don't really agree with that. I think that if someone wants to implement their own system, we should not impose them to have an attribute like `[IdPrefix("")]` that does nothing